api/v1alpha1/warehouse_types.go

```diff
@@ -510,6 +514,33 @@ type ChartSubscription struct {
 	DiscoveryLimit int32 `json:"discoveryLimit,omitempty" protobuf:"varint,4,opt,name=discoveryLimit"`
 }
 
+// GenericSubscription represents a subscription to something that is not a Git,
+// container image, or Helm chart repository.
+type GenericSubscription struct {
+	// ArtifactKind specifies the kind of artifact this subscription is for.
```

Comment:
I would love this to have a bit more documentation here as to what this string value should be. Is it entirely user defined? Is it a MIME type? I haven't gotten further so maybe it is in docs, but even so it might be worth clarifying here some more